### PR TITLE
fix couch blob attachments meta to have correct type code

### PR DESCRIPTION
**Pull request by jmtroth0**
For posterity: couch attachments were given the wrong type code in a migration awhile ago since the `_blobdb_type_code` for all xforminstancesql's is the one for `form_xml` https://github.com/dimagi/commcare-hq/blob/3788966119d1c63300279418a5bf2fc31ad37f6f/corehq/blobs/migrate.py#L371

@millerdev buddy: @kaapstorm

### CODE REVIEW

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 1]
```diff
@@ -564,15 +564,30 @@ def _migrate_form_attachments(sql_form, couch_form):
     attachments = []
     metadb = get_blob_db().metadb
     for name, blob in six.iteritems(couch_form.blobs):
+        def try_to_get_blob_meta(parent_id, type_code, name):
```
Comment (millerdev) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: nit: what do you think of moving this function definition out of the for-loop?

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 2]
```diff
@@ -564,15 +564,30 @@ def _migrate_form_attachments(sql_form, couch_form):
...
-        except BlobMeta.DoesNotExist:
+        meta = try_to_get_blob_meta(sql_form.form_id, type_code, name)
+
+        # there was a bug in a migration causing the type code for many form attachments to be set as form_xml
+        # this checks the db for a meta resembling this and fixes it for postgres
+        # https://github.com/dimagi/commcare-hq/blob/3788966119d1c63300279418a5bf2fc31ad37f6f/corehq/blobs/migrate.py#L371
+        if not meta and name != "form.xml":
+            meta = try_to_get_blob_meta(sql_form.form_id, CODES.form_xml, name)
+            if meta:
+                meta.type_code = CODES.form_attachment
```
Comment (millerdev) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: I think a `meta.save()` is needed after this line.

File: corehq/apps/couch_sql_migration/couchsqlmigration.py (hunk 1 above)
Comment (jmtroth0) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: e0c8763

File: corehq/apps/couch_sql_migration/couchsqlmigration.py (hunk 2 above)
Comment (jmtroth0) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: 289c619

**Review by millerdev — APPROVED** [review]

**Pull request merged**